diag/diagnostics_test.go

```diff
@@ -518,3 +519,123 @@ func TestDiagnosticsHasError(t *testing.T) {
 		})
 	}
 }
+
+func TestDiagnosticsErrorsCount(t *testing.T) {
+	t.Parallel()
+
+	type testCase struct {
+		diags    diag.Diagnostics
+		expected int
+	}
+	tests := map[string]testCase{
+		"errors": {
```

Comment:
Nit: Similar to the `Errors` testing, I think we should cover some additional cases just to ensure it returns `0` instead of panics, etc.

Reply:
Added additional test coverage.